packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.web.test.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  getBaseBoxStyles,
+  getDependencyProps,
+  getResponsiveValue,
+  getSpacingValue,
+  shouldAddBreakpoint,
+} from './getBaseBoxStyles';
+import type { BaseBoxProps } from './types';
+import paymentLightTheme from '~components/BladeProvider/__tests__/paymentLightTheme/paymentLightTheme';
+
+describe('getResponsiveValue', () => {
+  it('should return correctly for plain values', () => {
+    expect(getResponsiveValue('hello', 'base')).toBe('hello');
+    expect(getResponsiveValue('hello', 'xs')).toBe(undefined);
+  });
+
+  it('should correctly handle falsy values', () => {
+    expect(getResponsiveValue(undefined, 'base')).toBe(undefined);
+    // @ts-expect-error: intentional null to check the falsy values
+    expect(getResponsiveValue(null, 'base')).toBe(undefined);
+    expect(getResponsiveValue(0, 'base')).toBe(0);
+    expect(getResponsiveValue('', 'base')).toBe('');
+    expect(getResponsiveValue({ base: 3, s: 0 }, 's')).toBe(0);
+  });
+
+  it('should return the right responsive value for given size', () => {
+    expect(getResponsiveValue({ base: 'base-value', m: 'medium-value' }, 'm')).toBe('medium-value');
+    expect(getResponsiveValue({ base: 'base-value', m: 'medium-value' }, 'base')).toBe(
```

Comment:
- https://github.com/razorpay/blade/pull/1009#issuecomment-1447696941
- Yeah we can call it `breakpoint`. Will make the change
- Its type is the type we pass to object. So if I do `getResponsiveValue({ base: 'base-value', m: 'medium-value' }, 'm'))` then it will return one of `base-value` or `medium-value`. If I pass number, it will return number.

Reply:
+1, lets make it `breakpoint`, seems more intuitive